packages/worker/src/index.ts

```diff
@@ -5,60 +5,126 @@ import {
 	parseTranscriptJobMessage,
 	isFailure,
 	deleteMessage,
+	changeMessageVisibility,
+	TranscriptionConfig,
+	TranscriptionOutput,
 } from '@guardian/transcription-service-common';
 import { getSNSClient, publishTranscriptionOutput } from './sns';
+import {
+	getTranscriptionText,
+	convertToWav,
+	createContainer,
+} from './transcribe';
+import {
+	getFile,
+	getS3Client,
+} from '@guardian/transcription-service-common/src/s3';
+import path from 'path';
 
 const main = async () => {
-	const config = await getConfig();
-	const client = getClient(config.aws.region, config.aws.localstackEndpoint);
-
-	// to simulate a transcription job, delay 5 seconds in DEV, 2 minutes in PROD before deleting the message
-	const dummyDelay = config.app.stage === 'DEV' ? 5 : 120;
-	// override timeout to allow enough time for our dummy delay before message becomes available
-	const timeoutOverride = dummyDelay + 1;
-	const message = await getNextMessage(
-		client,
-		config.app.taskQueueUrl,
-		timeoutOverride,
-	);
+	try {
+		const config = await getConfig();
 
-	const snsClient = getSNSClient(
-		config.aws.region,
-		config.aws.localstackEndpoint,
-	);
+		const numberOfThreads = config.app.stage === 'PROD' ? 16 : 2;
 
-	if (isFailure(message)) {
-		return;
-	}
-	console.log(config.app.taskQueueUrl);
-
-	await publishTranscriptionOutput(
-		snsClient,
-		config.app.destinationTopicArns.transcriptionService,
-		{
-			id: 'test-id',
-			transcriptionSrt: 'test-srt',
-			languageCode: 'en',
-			userEmail: 'test@test.com',
-		},
-	);
+		const client = getClient(config.aws.region, config.aws.localstackEndpoint);
+		const message = await getNextMessage(client, config.app.taskQueueUrl);
+		const snsClient = getSNSClient(
+			config.aws.region,
+			config.aws.localstackEndpoint,
+		);
+
+		if (isFailure(message)) {
+			console.error(`Failed to fetch message due to ${message.errorMsg}`);
+			return;
+		}
+
+		if (!message.message) {
+			console.log('No messages available');
+			return;
+		}
 
-	if (message.message?.ReceiptHandle) {
 		const job = parseTranscriptJobMessage(message.message);
+
 		console.log(
 			`Fetched transcription job with id ${message.message.MessageId}}`,
 			job,
 		);
-		// wait for 3 minutes then delete the message to simulate the transcription
-		setTimeout(() => {
+
+		if (!job) {
+			console.error('Failed to parse job message', message);
+			return;
+		}
+
+		const fileToTranscribe = await getFileFromS3(config, job?.s3Key);
```

Comment:
at this point job isn't can't be undefined, right?
```suggestion
		const fileToTranscribe = await getFileFromS3(config, job.s3Key);
```